### ISSUE TITLE
Cannot compile demography.md with newest msprime

### ISSUE THREAD

**Issue by hyanwong** [tskit-dev/tutorials#55]
I'm getting

```
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
dem = msprime.Demography(populations=[pop0, pop1], migration_matrix=M, events=[migration_rate_change])
ts = msprime.sim_ancestry(samples=[samples0, samples1], demography=dem, sequence_length=1000, random_seed=63461, recombination_rate=1e-7)
------------------

ValueError                                Traceback (most recent call last)
<ipython-input-8-acd1a2c2edb0> in <module>
----> 1 dem = msprime.Demography(populations=[pop0, pop1], migration_matrix=M, events=[migration_rate_change])
ValueError: Population ID should not be set before using to create a Demography

```

**Comment by hyanwong** [comment]
I don't understand why this is, as the raw code seems to run fine for me in a notebook:

```
import msprime
import numpy as np
from IPython.display import SVG
pop0 = msprime.Population(initial_size=500, growth_rate=0, name='Population0')
pop1 = msprime.Population(initial_size=500, growth_rate=0, name='Population1')
samples0 = msprime.SampleSet(num_samples=3, population='Population0')
samples1 = msprime.SampleSet(num_samples=3, population='Population1')
M = np.array([
[0, 0.05],
[0.02, 0]])
dem = msprime.Demography(populations=[pop0, pop1], migration_matrix=M)
ts = msprime.sim_ancestry(samples=[samples0, samples1], demography=dem, sequence_length=1000, random_seed=17, recombination_rate=1e-7)
```

**Comment by jeromekelleher** [comment]
We need to update these examples as they aren't idiomatic (or correct, I think) - we don't create objects any more, but rather call methods like ``add_population`` etc. The above would be 
```python
dem = msprime.Demography()
dem.add_population(name="Population0", initial_size=500)
dem.add_population(name="Population1", initial_size=500)
# Not sure we'd get into asymmetric migration rates so soon, this is quite advanced. Anyway.
dem.set_migration_rate(source="Population0", dest="Population1", rate=0.05)
dem.set_migration_rate(source="Population0", dest="Population1", rate=0.02)
ts = msprime.sim_ancestry({"Population0": 3, "POpulation1": 3}, demography=demography, # etc)
```

**Comment by hyanwong** [comment]
Do you want me to do this or leave it to Georgia in a re-write? At the moment I suspect CI might fail when I make changes to other pages.

**Comment by jeromekelleher** [comment]
I'll leave that to you and @gtsambos.

**Comment by gtsambos** [comment]
updating now 👍

**Comment by jeromekelleher** [comment]
Closed in #57
